nightfall-optimist/src/services/check-block.mjs

```diff
@@ -240,13 +240,13 @@ async function checkDuplicateNullifiersWithinBlock(block, transactions) {
  */
 // eslint-disable-next-line import/prefer-default-export
 export async function checkBlock(block, transactions) {
-  await checkLeafCount(block);
-  // now we have to check the commitment root.
-  // For this we can make use of Timber with its optimistic extensions.
-  await checkBlockRoot(block);
-  await checkFrontier(block);
-  await checkDuplicateCommitmentsWithinBlock(block, transactions);
-  await checkDuplicateNullifiersWithinBlock(block, transactions);
+  await Promise.all([
+    await checkLeafCount(block),
```

Comment:
Hey @pawelgrzybek, I'm not expert in JS, but isn't this better?
```
await Promise.all([
    checkLeafCount(block),
    checkBlockRoot(block),
    checkFrontier(block),
    checkDuplicateCommitmentsWithinBlock(block, transactions),
    checkDuplicateNullifiersWithinBlock(block, transactions),
  ]);
```

Reply:
Sorry, so what exactly is the difference between yours and mine?